### ISSUE TITLE
getFullyQualifiedName returns normal name, not fully qualified one

### ISSUE THREAD

**Issue by atasak** [dsherret/ts-morph#132]
When I used the fully qualified names to match up classes, I noticed that getFullyQualifiedName does not always include the path. I encountered the bug when: 

`export class A {
    str: string = 'Hello World';
    readonly num: number = 10;

    method(x: string, y: number, b: B): boolean {
        return false;
    }
}

class B extends A {
    str: string = 'Nope';
    y: string[];
    a: A;
}`

- Read the file
- Iterate over classes with `source.getClasses()` (it will encounter class `A`)
- Iterate over methods with `clazz.getInstanceMethods()` (it will encounter method `method`)
- Iterate over the parameters with `method.getParameters()` (it will encounter parameter `b: B`)
- Get the type with `parameter.getType()`
- Get the symbol with `type.getSymbol()`
- Get the name with `symbol.getFullyQualifiedName()`

In short: 
`source.getClasses()[0].getInstanceMethods()[0].getParameters()[2].getType().getSymbol().getFullyQualifiedName()`

**Comment by dsherret** [comment]
Hi @atasak, so this is getting the fully qualified name of `B`? I believe the output should be `B`? What's the actual output?

**Comment by atasak** [comment]
Hmmm, I may have misunderstood how fully qualified names are computed. I expected the fully qualified name of B to be like A: `"C:/Users/niekj/Documents/TsOne/ApiOne/spec/common/tests".A`. With B, it's just `B`. And after a full recompilation the errors I received because of it are gone...

Will the fully qualified name of non-exported members always be without the path? If I have a non-exported type B in another file, how can I differentiate between them? I kind of need all classes to have a unique name, even though they may have the same name and not be exported.

**Comment by dsherret** [comment]
The result of `getFullyQualifiedName` is what TypeScript's underlying TypeChecker returns when calling its `getFullyQualifiedName(symbol: Symbol): string;` method. I can see why they might have decided not to also include the path for non-exported member's fully qualified name (because they're not accessible via that path).

I'm not entirely sure what problem you're trying to solve, but I guess what you could do is append the path yourself:

```typescript
// untested
let fullyQualifiedName = symbol.getFullyQualifiedName();
if (!fullyQualifiedName.startsWith(`"`)) {
    const filePath = symbol.getDeclarations()[0].sourceFile.getFilePath();
    fullyQualifiedName = `"${filePath}".${fullyQualifiedName}`;
}
```

**Comment by atasak** [comment]
It works with `getSourceFile()` instead of `sourceFile`. Thanks!
